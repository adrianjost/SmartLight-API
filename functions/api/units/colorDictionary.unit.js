const {
	extractPercentage,
	AdjustColorBrightness,
	AdjustColorTemperature,
	lighten,
	darken,
	heatUp,
	coolDown,
	list,
} = require("./colorDictionary");

describe("api/units/colorDictionary", () => {
	it("exports a list of colors", () => {
		expect(list.length > 0).toBe(true);
		list.forEach((entry) => {
			expect(entry.name).toStrictEqual(expect.any(String));
			expect([String, Function]).toContain(entry.value.constructor);
		});
	});

	describe("extractPercentage", () => {
		const prefix = "mache das Licht vom Bett";
		it("extract 0", () => {
			expect(extractPercentage(prefix + "0 Prozent heller")).toMatchSnapshot();
		});
		it("extract a positive Number", () => {
			expect(extractPercentage(prefix + "20 Prozent heller")).toMatchSnapshot();
		});
		it("can parse relative values", () => {
			expect(extractPercentage(prefix + "ein wenig heller")).toMatchSnapshot();
			expect(extractPercentage(prefix + "etwas heller")).toMatchSnapshot();
			expect(extractPercentage(prefix + "viel heller")).toMatchSnapshot();
		});
	});

	describe("AdjustColorBrightness", () => {
		it.each([
			["check overflows", "#000000", -20],
			["check overflows", "#ffffff", +20],
			["lighten black ", "#000000", +20],
			["darken white", "#ffffff", -20],
			["lighten single color", "#ff0000", +20],
			["darken single color", "#ff0000", -20],
		])("%s - args: %s, %i%%", (description, color, percentage) => {
			expect(AdjustColorBrightness(color, percentage)).toMatchSnapshot();
		});
	});

	describe("AdjustColorTemperature", () => {
		it.each([
			["check overflows", "#0000ff", -20],
			["check overflows", "#ff0000", +20],
			["heatup black ", "#000000", +20],
			["cool down white", "#ffffff", -20],
			["cool down warm white", "#ff0000", -20],
			["heat up cold white", "#000000ff", -20],
		])("%s - args: %s, %i%%", (description, color, percentage) => {
			expect(AdjustColorTemperature(color, percentage)).toMatchSnapshot();
		});
	});
});
