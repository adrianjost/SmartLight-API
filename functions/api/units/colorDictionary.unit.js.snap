// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`api/units/colorDictionary AdjustColorBrightness check overflows - args: #000000, -20% 1`] = `"#000000"`;

exports[`api/units/colorDictionary AdjustColorBrightness check overflows - args: #ffffff, 20% 1`] = `"#FFFFFF"`;

exports[`api/units/colorDictionary AdjustColorBrightness darken single color - args: #ff0000, -20% 1`] = `"#CC0000"`;

exports[`api/units/colorDictionary AdjustColorBrightness darken white - args: #ffffff, -20% 1`] = `"#CCCCCC"`;

exports[`api/units/colorDictionary AdjustColorBrightness lighten black  - args: #000000, 20% 1`] = `"#333333"`;

exports[`api/units/colorDictionary AdjustColorBrightness lighten single color - args: #ff0000, 20% 1`] = `"#FF0000"`;

exports[`api/units/colorDictionary AdjustColorTemperature check overflows - args: #0000ff, -20% 1`] = `"#0000ff"`;

exports[`api/units/colorDictionary AdjustColorTemperature check overflows - args: #ff0000, 20% 1`] = `"#ff0000"`;

exports[`api/units/colorDictionary AdjustColorTemperature cool down warm white - args: #ff0000, -20% 1`] = `"#e6001a"`;

exports[`api/units/colorDictionary AdjustColorTemperature cool down white - args: #ffffff, -20% 1`] = `"#e6ffff"`;

exports[`api/units/colorDictionary AdjustColorTemperature heat up cold white - args: #000000ff, -20% 1`] = `"#00001a"`;

exports[`api/units/colorDictionary AdjustColorTemperature heatup black  - args: #000000, 20% 1`] = `"#1a0000"`;

exports[`api/units/colorDictionary extractPercentage can parse relative values 1`] = `10`;

exports[`api/units/colorDictionary extractPercentage can parse relative values 2`] = `10`;

exports[`api/units/colorDictionary extractPercentage can parse relative values 3`] = `50`;

exports[`api/units/colorDictionary extractPercentage extract 0 1`] = `0`;

exports[`api/units/colorDictionary extractPercentage extract a positive Number 1`] = `20`;
